addons/longhorn/template/base/install.sh

```diff
@@ -119,3 +122,33 @@ function longhorn_preflight() {
     local src="${DIR}/addons/longhorn/${LONGHORN_VERSION}"
     echo "${src}/host-preflight.yaml"
 }
+
+function check_mount_propagation() {
+    local src=$1
+
+    kubectl get ns longhorn-system >/dev/null 2>&1 || kubectl create ns longhorn-system >/dev/null 2>&1
+
+    render_yaml_file "$src/tmpl-mount-propagation.yaml" > "$src/mount-propagation.yaml"
+    kubectl apply -f "$src/mount-propagation.yaml"
+    echo "Waiting for the Longhorn Mount Propagation Check Daemonset to be ready"
+    spinner_until 120 longhorn_daemonset_is_ready longhorn-manager
+
+    validate_longhorn_ds
+
+    kubectl delete -f "$src/mount-propagation.yaml"
+}
+
+function validate_longhorn_ds() {
+    local allSupported=true
+
+    local allpods=$(kubectl get daemonsets -n longhorn-system longhorn-environment-check --no-headers | tr -s ' ' | cut -d ' ' -f4)
+    local bidirectional=$(kubectl get pods -l app=longhorn-environment-check -o=jsonpath='{.items[*].spec.containers[0].volumeMounts[*]}' | grep -o 'Bidirectional' | wc -l)
+
+    if [ "$bidirectional" -lt "$allpods" ]; then
```

Comment:
this will fail with an error if allpods is empty

Reply:
That would show up in longhorn_daemonset_is_ready first, though